clang/lib/Driver/ToolChains/Repo.cpp

```diff
@@ -0,0 +1,195 @@
+
+using namespace clang::driver;
+using namespace clang::driver::tools;
+using namespace clang::driver::toolchains;
+using namespace clang;
+using namespace llvm::opt;
+
+static void constructRepoLinkArgs(Compilation &C, const JobAction &JA,
+                                  const toolchains::RepoMuslToolChain &RTC,
+                                  const InputInfo &Output,
+                                  const InputInfoList &Inputs,
+                                  const ArgList &Args, ArgStringList &CmdArgs,
+                                  const char *LinkingOutput) {
+
+  const Driver &D = RTC.getDriver();
+
+  //----------------------------------------------------------------------------
+  //
+  //----------------------------------------------------------------------------
+  CmdArgs.push_back("-o");
+  CmdArgs.push_back(Output.getFilename());
+
+  std::string MuslRoot = D.SysRoot.empty() ? "/usr/local/musl" : D.SysRoot;
+  if (!Args.hasArg(options::OPT_nostartfiles, options::OPT_nostdlib)) {
+    CmdArgs.push_back(Args.MakeArgString(MuslRoot + "/lib/crt1.t"));
+    CmdArgs.push_back(Args.MakeArgString(MuslRoot + "/lib/crt1_asm.t"));
+    CmdArgs.push_back(Args.MakeArgString(MuslRoot + "/lib/libc_repo.a"));
+  }
+
+  auto CRTPath = RTC.getCompilerRTPath();
+  if (!RTC.getVFS().exists(CRTPath)) // Build compiler-rt as stand-alone
+    CRTPath = "/usr/lib/linux";
+  if (!Args.hasArg(options::OPT_nostartfiles, options::OPT_nostdlib)) {
+    CmdArgs.push_back(
+        Args.MakeArgString(CRTPath + "/clang_rt.crtbegin-x86_64.o"));
+    CmdArgs.push_back(
+        Args.MakeArgString(CRTPath + "/clang_rt.crtend-x86_64.o"));
+  }
+
+  //----------------------------------------------------------------------------
+  // Library Search Paths
+  //----------------------------------------------------------------------------
+  CmdArgs.push_back(Args.MakeArgString(StringRef("-L") + MuslRoot + "/lib"));
+
+  if (RTC.getVFS().exists(CRTPath))
+    CmdArgs.push_back(Args.MakeArgString("-L" + CRTPath));
+
+  const ToolChain::path_list &LibPaths = RTC.getFilePaths();
+  for (const auto &LibPath : LibPaths)
+    CmdArgs.push_back(Args.MakeArgString(StringRef("-L") + LibPath));
+
+  //----------------------------------------------------------------------------
+  //
+  //----------------------------------------------------------------------------
+  Args.AddAllArgs(CmdArgs,
+                  {options::OPT_T_Group, options::OPT_e, options::OPT_s,
+                   options::OPT_t, options::OPT_u_Group});
+  AddLinkerInputs(RTC, Inputs, Args, CmdArgs, JA);
+
+  //----------------------------------------------------------------------------
+  // Libraries
+  //----------------------------------------------------------------------------
+  if (!Args.hasArg(options::OPT_nostdlib, options::OPT_nodefaultlibs)) {
+    CmdArgs.push_back("-lclang_rt.builtins-x86_64");
+  }
+
+  if (D.CCCIsCXX()) {
+    if (RTC.ShouldLinkCXXStdlib(Args))
```

Comment:
Combine the two if statements:
```cxx
if (D.CCCIsCXX() && RTC.ShouldLinkCXXStdlib(Args)) {}
```

Reply:
Fixed in the commit [0a629b99](https://github.com/SNSystems/llvm-project-prepo/pull/158/commits/0a629b99bb27274ce786c619db54e1a59d902947).